bookkeeper-server/src/main/java/org/apache/bookkeeper/client/impl/LedgerEntriesImpl.java

```diff
@@ -0,0 +1,104 @@
+/*
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.client.impl;
+
+import io.netty.util.Recycler;
+import java.util.Iterator;
+import java.util.List;
+import org.apache.bookkeeper.client.api.LedgerEntry;
+
+public class LedgerEntriesImpl implements org.apache.bookkeeper.client.api.LedgerEntries {
+    private List<LedgerEntry> entries;
+    private final Recycler.Handle<LedgerEntriesImpl> recyclerHandle;
+
+    private LedgerEntriesImpl(Recycler.Handle<LedgerEntriesImpl> recyclerHandle) {
+        this.recyclerHandle = recyclerHandle;
+    }
+
+    private static final Recycler<LedgerEntriesImpl> RECYCLER = new Recycler<LedgerEntriesImpl>() {
+        @Override
+        protected LedgerEntriesImpl newObject(Recycler.Handle<LedgerEntriesImpl> handle) {
+            return new LedgerEntriesImpl(handle);
+        }
+    };
+
+    private void recycle() {
+        releaseByteBuf();
+        recyclerHandle.recycle(this);
+    }
+
+    private void releaseByteBuf() {
+        if (entries != null) {
+            for (LedgerEntry entry : entries) {
+                entry.getEntryBuffer().release(1);
+            }
+        }
+    }
+
+    /**
+     * Create ledger entries.
+     *
+     * @param entries the entries with ordering
+     * @return the LedgerEntriesImpl
+     */
+    public static LedgerEntriesImpl create(List<LedgerEntry> entries) {
```

Comment:
checkArgument(!entries.isEmpty(), "...")

line 73 assumes entries is not empty.

Reply:
Thanks. will change it.